### PR TITLE
drivers: flash: nRF: Move sync ticker to Subsys Bluetooth Controller

### LINKED ISSUE #55756: Bluetooth: Controller: Move soc_flash_nrf_ticker into `subsys/bluetooth/controller/flash`
> The code is also using private header files from `subsys/bluetooth/controller`, which would speak more in favour of having this code under there too.

ok. We can have the coex driver as `subsys/bluetooth/controller/coex/coex_nrf_ticker.c`. Subsequently move https://github.com/zephyrproject-rtos/zephyr/blob/75bb8ce3826fabe55acb52b18de7116f8a206deb/drivers/flash/soc_flash_nrf_ticker.c into `subsys/bluetooth/controller/flash/flash_nrf_ticker.c`  in a different PR. This way we will have `hci`, `flash` and `coex` which include private header files be in `subsys/bluetooth/controller/{hci,flash,coex}`

_Originally posted by @cvinayak in https://github.com/zephyrproject-rtos/zephyr/issues/51419#issuecomment-1295228056_

### PR DESCRIPTION
**Pull request by cvinayak**
Move the SoC Flash nRF sync ticker implementation into Bluetooth Controller Subsystem folder, as internal headers are included.

Refactor the Bluetooth Controller's CMakelists.txt to consistently use zephyr_library_sources_ifdef() and zephyr_library_include_directories_ifdef().

Fixes #55756

### CODE REVIEW

File: subsys/bluetooth/controller/CMakeLists.txt [hunk 1]
```diff
@@ -1,179 +1,194 @@
...
+    CONFIG_BT_CTLR_ADV_PERIODIC
+    ll_sw/ull_adv_sync.c
+    )
+  zephyr_library_sources_ifdef(
+    CONFIG_BT_CTLR_ADV_ISO
+    ll_sw/ull_adv_iso.c
+    )
+endif()
 
-if(CONFIG_BT_LL_SW_SPLIT)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/CMakeLists.txt:22]: i should not remove this!

File: subsys/bluetooth/controller/CMakeLists.txt [hunk 2]
```diff
@@ -1,179 +1,194 @@
 # SPDX-License-Identifier: Apache-2.0
 
+if(CONFIG_BT_CTLR_ADVANCED_FEATURES)
```
Comment (kruithofa) [inline subsys/bluetooth/controller/CMakeLists.txt:3]: is it possible to make the refactoring a separate commit from the move of the nrf sync ticker?

File: subsys/bluetooth/controller/CMakeLists.txt (hunk 1 above)
Comment (kruithofa) [reply subsys/bluetooth/controller/CMakeLists.txt:22]: from subsys/bluetooth/CMakeLists.txt:

```
if(CONFIG_BT_CTLR AND CONFIG_BT_LL_SW_SPLIT)
  add_subdirectory(controller)
endif()
```

so yes, removing this should be ok

**Review by jori-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
